packages/mongo-nav/src/mongo-select/OrgSelect.tsx

```diff
@@ -0,0 +1,363 @@
+  justify-content: center;
+  border-radius: 0 5px 5px 0;
+  border: 1px solid ${uiColors.gray.light2};
+  border-left: 0;
+  background-color: white;
+  outline: none;
+  transition: all 150ms ease-in-out;
+
+  ${facepaint({
+    marginRight: ['16px', '16px', '20px'],
+    height: ['30px', '36px', '36px'],
+    paddingRight: ['6px', '8px', '8px'],
+    paddingLeft: ['6px', '8px', '8px'],
+  })}
+
+  &:hover {
+    background-color: ${uiColors.gray.light2};
+    border-color: ${uiColors.gray.light2};
+    color: ${uiColors.gray.dark2};
+  }
+`;
+
+const orgTriggerContainerStyle = css`
+  position: relative;
+  z-index: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  margin-left: 20px;
+`;
+
+const orgTriggerRingStyle = css`
+  border-radius: 7px 0 0 7px;
+`;
+
+const orgOptionContainerStyle = css`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const viewAllStyle = css`
+  color: ${uiColors.blue.base};
+`;
+
+const productStyle = css`
+  font-size: 12px;
+  color: ${uiColors.gray.dark2};
+  font-weight: bolder;
+  white-space: nowrap;
+`;
+
+const emptyStateStyle = css`
+  font-size: 14px;
+  padding: 4px 8px;
+  margin-bottom: 20px;
+`;
+
+const linkStyle = css`
+  color: ${uiColors.blue.base};
+  text-decoration: none;
+  &:visited {
+    color: ${uiColors.blue.base};
+    text-decoration: none;
+  }
+`;
+
+const focusedLinkStyle = css`
+  &:focus {
+    background-color: ${uiColors.blue.light2};
+    color: ${uiColors.blue.dark2};
+    border-color: ${uiColors.blue.light2};
+  }
+`;
+
+const activeIconStyle = css`
+  color: ${uiColors.green.base};
+`;
+
+const disabledTextStyle = css`
+  color: ${uiColors.gray.light1};
+`;
+
+// types
+interface OrganizationMongoSelectProps extends BaseMongoSelectProps {
+  data?: Array<OrganizationInterface>;
+  current?: CurrentOrganizationInterface;
+  constructOrganizationURL: NonNullable<
+    MongoNavInterface['constructOrganizationURL']
+  >;
+  isOnPrem?: boolean;
+  isActive?: boolean;
+  disabled?: boolean;
+}
+
+const formattedPlanTypes: Record<PlanType, string> = {
+  [PlanType.Atlas]: 'Atlas',
+  [PlanType.Cloud]: 'Cloud Manager',
+  [PlanType.OnPrem]: 'Ops Manager',
+} as const;
+
+// data props
+const triggerDataProp = createDataProp('org-trigger');
+const anchorDataProp = createDataProp('anchor-data-prop');
+
+function OrgSelect({
+  current,
+  data = [],
+  urls = {},
+  onChange: onChangeProp,
+  constructOrganizationURL,
+  isOnPrem = false,
+  isActive = false,
+  disabled = false,
+  loading = false,
+}: OrganizationMongoSelectProps) {
+  const [value, setValue] = useState('');
+  const [open, setOpen] = useState(false);
+  const onElementClick = useOnElementClick();
+  const { usingKeyboard: showFocus } = useUsingKeyboardContext();
+  const { width: viewportWidth } = useViewportSize();
+  const isTablet = viewportWidth < breakpoints.medium;
+
+  const toggleOpen = () => {
+    setOpen(!open);
+    if (!open) {
+      setValue('');
```

Comment:
Why the change?

Reply:
This previously used `useEffect` and split handling the open and value states over the trigger and the select. Now that they're integrated, this would update its own state instead, and was causing an infinite loop of renders.